keadm/cmd/keadm/app/cmd/edge/batchjoin.go

```diff
@@ -0,0 +1,179 @@
+			if err := executeJoinCommand(ctx, &n, &config.KeadmParams); err != nil {
+				fmt.Printf("Processing node %s  executeJoinCommand error \n", n.IP)
+				errCh <- err // Send to the error channel when an error occurs
+				return
+			}
+			fmt.Printf("Node %s joined successfully\n", n.IP)
+		}(node)
+	}
+
+	// Wait for all goroutines to complete
+	wg.Wait()
+	close(errCh)
+
+	// Check for errors
+	for err := range errCh {
+		if err != nil {
+			return err // 如果有任意一个错误，返回第一个错误
+		}
+	}
+
+	return nil // All nodes processed successfully
+}
+
+func downloadAndCopyKeadmToNode(ctx context.Context, node *Node) error {
+	// Download keadm from GitHub Release
+	releaseURL := fmt.Sprintf("https://github.com/kubeedge/kubeedge/releases/download/%s/keadm-%s.tar.gz", node.KubeedgeVersion, node.Architecture)
+	fmt.Println("releaseURL:", releaseURL)
+	// SSH connect to the node
+	client, err := createSSHClient(node.Username, node.IP, node.Password)
+	if err != nil {
+		return fmt.Errorf("failed to connect to node %s: %v", node.IP, err)
+	}
+	defer client.Close()
+
+	// Execute remote command to download keadm
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create new SSH session: %v", err)
+	}
+	defer session.Close()
+	fileName := fmt.Sprintf("keadm-%s.tar.gz", node.Architecture)
+	cmd := fmt.Sprintf("curl -L %s -o /tmp/%s", releaseURL, fileName)
+	if err := session.Run(cmd); err != nil {
+		return fmt.Errorf("failed to download keadm on node %s: %v", node.IP, err)
+	}
+	return nil
+}
+
+func executeJoinCommand(ctx context.Context, node *Node, params *KeadmParams) error {
+	// SSH connect to the node
+	client, err := createSSHClient(node.Username, node.IP, node.Password)
+	if err != nil {
+		return fmt.Errorf("failed to connect to node %s: %v", node.IP, err)
+	}
+	defer client.Close()
+
+	// Execute the keadm join command
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create new SSH session: %v", err)
+	}
+	defer session.Close()
+	fileName := fmt.Sprintf("keadm-%s.tar.gz", node.Architecture)
+	dirName := fmt.Sprintf("keadm-%s", node.Architecture)
+	cmd := fmt.Sprintf("cd /tmp && tar xzf %s && cd %s/keadm && ./keadm join --cloudcore-ipport=%s --token=%s --kubeedge-version=%s --cgroupdriver systemd --runtimetype=docker",
+		fileName, dirName, params.CloudCoreIPPort, params.Token, params.KubeedgeVersion)
+	fmt.Println("cmd:", cmd)
+	if err := session.Run(cmd); err != nil {
+		return fmt.Errorf("failed to execute keadm join command on node %s: %v", node.IP, err)
+	}
+	return nil
+}
+
+func createSSHClient(username, host, password string) (*ssh.Client, error) {
+	// Create SSH configuration
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	// Connect to the SSH server
+	client, err := ssh.Dial("tcp", fmt.Sprintf("%s:22", host), config)
```

Comment:
Suggest allowing support for custom ssh port, as the default 22 port is usually not used in production environments

Reply:
+1

Reply:
Thank you for your suggestions and I apologize for responding now, I will make specific changes based on your suggestions.

Reply:
> Suggest allowing support for custom ssh port, as the default 22 port is usually not used in production environments



> +1

Thank you for your suggestions and I apologize for responding now, I will make specific changes based on your suggestions.